server/service/src/requisition/program_settings/mod.rs

```diff
@@ -0,0 +1,322 @@
+mod map;
+mod prepare;
+
+use map::map;
+use prepare::prepare;
+
+use repository::{
+    PeriodRow, ProgramRequisitionOrderTypeRow, ProgramRequisitionSettings, ProgramSupplier,
+    RepositoryError,
+};
+
+use crate::service_provider::ServiceContext;
+
+#[derive(Debug, PartialEq)]
+pub struct OrderType {
+    pub order_type: ProgramRequisitionOrderTypeRow,
+    pub available_periods: Vec<PeriodRow>,
+}
+
+#[derive(Debug, PartialEq)]
+pub struct ProgramSettings {
+    pub program_requisition_settings: ProgramRequisitionSettings,
+    pub suppliers: Vec<ProgramSupplier>,
+    pub order_types: Vec<OrderType>,
+}
+
+// TODO only return and calculate for X number of periods from now ? (back and forward)
+
+/// Method will calculate program settings for a store, broken down into two tasks, prepare and map
+/// See prepare and map for detailed descriptions
+pub(super) fn get_program_requisition_settings(
+    ctx: &ServiceContext,
+    store_id: &str,
+) -> Result<Vec<ProgramSettings>, RepositoryError> {
+    // Get program_settings, order_types, periods and requisitions_in_periods for a store
+    let Some(prepared) =
+        prepare(ctx, store_id)? else {
+            return Ok(Vec::new())
+        };
```

Comment:
I'm not used to this syntax. What do you see as the advantage over this?
(which feels easier to understand in my mind)
```suggestion
    let prepared = match prepare(ctx, store_id) {
        Ok(Some(prepared)) => prepared,
        Ok(None) => return Ok(vec![]),
        Err(e) => return Err(e),
    };
```

Reply:
> I'm not used to this syntax.

It's early return, at the same time as matching a variant

> What do you see as the advantage over this?

I think it's stylistic preference really, I find it quite clean, is it the `?` that's confusing ?

We used to do this btw, you can see it in validation methods in service
```rust
let prepared = match prepare(ctx, store_id)? {
        None => return Ok(vec![]),
        Some(prepared) => prepared,
    };
```